src/fmu/dataio/_utils.py

```diff
@@ -174,22 +173,10 @@ def export_file_compute_checksum_md5(
     filename: Path,
     extension: str,
     flag: str | None = None,
-    tmp: bool = False,
-) -> tuple[Path | None, str]:
-    """Export and compute checksum, with possibility to use a tmp file."""
-
-    usefile: Path | None = filename
-    if tmp:
-        tmpdir = tempfile.TemporaryDirectory()
-        usefile = Path(tmpdir.name) / "tmpfile"
-
-    assert usefile is not None
-    export_file(obj, usefile, extension, flag=flag)
```

Comment:
If `export_file(...)` fails, and we are using a tmp file it wont get cleaned up. The new code will do so by usage of the context manger.